File: js/dailies/04_aug_18/binarySearch.js
const binarySearch = (arr, x) => {
	let begIdx = 0;
	let endIdx = arr.length - 1;

	while (begIdx <= endIdx) {
		let mid = Math.floor(endIdx - begIdx / 2) + begIdx;

		if (arr[mid] === x) {
			return true;
		} else if (arr[mid] < x) {
			begIdx = mid + 1;
		} else {
			endIdx = mid - 1;
		}
	}

	return false;
};

console.log(binarySearch([0, 3, 5, 6, 7, 8, 9], 10));

File: python/dailies/29_oct_22/binary_search.py
import math

from typing import List

from helpers.testing.list_search_tester import ListSearchTester


def binary_search(arr: List[int], x: int) -> int:
    beg_idx = 0
    end_idx = len(arr) - 1

    while beg_idx <= end_idx:
        mid = math.floor((end_idx - beg_idx) / 2) + beg_idx

        if arr[mid] == x:
            return True

        if arr[mid] < x:
            beg_idx = mid + 1
        else:
            end_idx = mid - 1

    return False


ListSearchTester(binary_search)
